docs/source/ops/index.rst

```diff
@@ -0,0 +1,86 @@
+Other limitations
+-----------------
+
+- If the number of divulged contracts continually grows, the storage used by the participant node may still keep growing, even with regular pruning.
+- Pruning may be rejected even if the node is running correctly (for example, to preserve non-repudiation properties); in this case, the application might not be able to archive contracts containing PII or pruning of these contracts may not be possible; thus, actually deleting this PII may also be technically unfeasible.
+- Pruning may leave parties, packages, and configuration data on the participant node, even if they are no longer needed for transaction processing, and even if they contain PII [3]_.
+- Pruning does not move pruned information to cold storage but simply deletes pruned data; for this reason, it is advisable to back up the Participant Index DB before invoking pruning. See the next sub-section for more Participant Index DB-related advice before and after invoking `prune`.
+- Pruning is not selective but rather effectively truncates the ledger, removing the transactions and archived contracts at and before the pruning offset.
+
+.. [3] This might be improved in future versions.
+
+Pruning and the DAML Participant Index DB
+-----------------------------------------
+
+Activities to be carried out *before* invoking a pruning operation include:
+
+- Backing up the Participant Index DB, as pruning will not move information to cold storage but rather it will delete archived contracts and transactions earlier than or at the pruning offset.
+
+Activities to be carried out *after* invoking a pruning operation include:
+
+- On a PostreSQL Index DB, a manual `VACUUM FULL` command is needed for the DB to give back disk space to the OS.
+
+Determining a suitable pruning offset
+-------------------------------------
+
+The :ref:`Active Contract Service <active-contract-service>` and the :ref:`Transaction Service <transaction-service>` provide offset information for transactions and for Active Contracts snapshots respectively: such offset can be used unchanged with `prune` calls.
+
+Scheduled jobs, applications and/or operator tools can be built on top of the DAML Ledger API to implement pruning automatically, for example at regular intervals, or on-demand, for example according to a user-initiated process.
+
+For instance, pruning at regular intervals could be performed by a cron job that:
+
+1. If a pruning interval has been saved to a well-known location:
+   a. Backs up the DAML Participant Index DB.
+   b. Performs pruning.
+   c. (If using PostgreSQL) Performs a `VACUUM FULL` command on the DAML Participant Index DB.
+
+2. Acquires a fresh Active Contract Set and saves the offset.
+
+Pruning could also be initiated on-demand at the offset of a specific transaction [4]_, for example as provided by a user application based on a search.
+
+.. [4] Note that not only a specific transaction but also earlier transactions and archived contracts will be pruned.
```

Comment:
The approach described here to determine an offset is a bit vague, and leaves quite a bit of work to the operator. Can we do better?

I mean with access to the index-db it is a simple query to find an appropriate offset. Can, should, do we expose such information on the Ledger API?

Reply:
I'd avoid making schema-dependent suggestions, so I'd rather support it at the API level as a feature improvement (and document it then). Not sure if we can improve the present suggestions about determining an offset without that; @meiersi-da @gerolf-da WDYT?

Reply:
What do you mean by "such information? Transactions have offsets and a record time. We currently don't have a way to scan backwards until you reach an offset that has a certain age.
While one can do that by accessing the index db directly, this won't be very convenient for operational tooling that might not have access to the DB.
If we assume (maybe incorrectly?) that usually users would want a rolling pruning window of x days, then maybe an additional service method that allows for querying the nearest offset to a given timestamp would allow for such operational tooling to be efficiently built with the ledger api. WDYT, @meiersi-da, @fabiotudone-da?

Reply:
Sorry, didn't see your comment while I was typing mine. I'm in favor of of an additional Ledger API endpoint.

Reply:
The original participant pruning design also allowed to provide a timestamp in `PruneRequest`. Maybe we should pick this up again and return the pruning offset in the response? It would avoid the extra round trip and seemlessly support the rolling pruning window use case.

Reply:
I agree that conveniently augmenting the Pruning Ledger API is the ultimate goal but operational tooling could be a good intermediate step and it doesn't need to be tied to (or communicating with) the DB directly, for example it could leverage an internal admin API. Or we could expose experimental APIs although I am a bit wary of the de-facto dependencies that we'd allow in that case.

Another option, after discussing with @oliverse-da (thanks!), would be to have the operator determine the offset by adding a specific amount to either the ledger's start, if not pruned, or to the most recent available offset returned as part of the error message when querying about the ledger's start (and, to avoid exposing the message format, we could make the offset within available through tooling). That amount would be sized based on the deployment's events rate.

It looks like all improved suggestions require some additional development though, so I'd leave this part of the doc as it is for now.

I'd also suggest to keep involving @oliverse-da and team (subject to their availability, of course) for the upcoming improvements and developments as they have done a great deal of thinking, design, experiments, measurements and have overall a lot of experience with pruning towards significant production-ready deployments.

Reply:
For multi-domain ledgers time will not necessarily map unambiguously to offsets as time does not grow monotonically wrt offsets. (cc @oggy- ) - see my note [above](https://github.com/digital-asset/daml/pull/8138/files#r535989613) for how we currently fill the gap until we have a ledger-api-way.

Reply:
@oliverse-da I replaced "earlier" occurrences with "at previous offsets" or similar. Will this do?

Reply:
> @oliverse-da I replaced "earlier" occurrences with "at previous offsets" or similar. Will this do?

Sorry Fabio, my reply in the previous suggestion was in response to Gerolf's reference to the old "PruneByTime" flavor we used to support in canton, but had gotten rid of because of time-semantic reasons. I did not intend for you to rephrase "earlier" with "at previous offsets" although that seems good.